src/ucp/core/ucp_worker.c

```diff
@@ -1188,6 +1189,31 @@ unsigned ucp_worker_progress(ucp_worker_h worker)
     return count;
 }
 
+ssize_t ucp_stream_worker_poll(ucp_worker_h worker,
+                               ucp_stream_poll_ep_t *poll_eps,
+                               size_t max_eps, unsigned flags)
+{
+    ssize_t count = 0;
+
+    UCP_THREAD_CS_ENTER_CONDITIONAL(&worker->mt_lock);
+
+    while ((count < max_eps) && !ucs_list_is_empty(&worker->stream_eps)) {
+        struct ucp_ep_ext_stream *ep_stream =
+            ucs_list_extract_head(&worker->stream_eps,
+                                  struct ucp_ep_ext_stream, list);
+        ucp_ep_h ep = ep_stream->ucp_ep;
+        ep->flags &= ~UCP_EP_FLAG_STREAM_IS_QUEUED;
```

Comment:
Where is it set? Will it be a part of next PR?

Reply:
Yes, i'll add this in next PR, and testing in another